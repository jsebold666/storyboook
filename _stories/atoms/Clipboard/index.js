import React from 'react';
import Clipboard from '../../../components/atoms/Clipboard';
import readme from './README.md';

const getText = () => 'Texto copiado para o clipboard...';

const component = () => <Clipboard getText={getText}>Clique aqui para copiar</Clipboard>;

export default [readme, component];
