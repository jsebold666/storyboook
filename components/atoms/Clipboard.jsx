import React, { Component } from 'react';
import clipboard from 'clipboard-polyfill';

export default class Clipboard extends Component {
  constructor(props) {
    super(props);

    this.state = {
      copiedText: false
    };
  }

  copy() {
    clipboard
      .writeText(this.props.getText())
      .then(
        function() {
          this.setState({
            copiedText: true
          });
        }.bind(this)
      )
      .catch(
        function() {
          this.setState({
            copiedText: false
          });
        }.bind(this)
      );
  }

  render() {
    return (
      <div {...this.props} onClick={this.copy.bind(this)}>
        {this.props.children}
      </div>
    );
  }
}
