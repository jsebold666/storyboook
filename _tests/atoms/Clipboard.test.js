/* globals mount */
import React from 'react';
import Clipboard from '../../components/atoms/Clipboard';

describe('Expect <Clipboard>', () => {
  it('to render', () => {
    const props = {
      value: 'This is text',
      onChange: () => {}
    };

    const wrapper = mount(<Clipboard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('to render properly with empty', () => {
    const props = {
      value: '',
      onChange: () => {}
    };

    const wrapper = mount(<Clipboard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
