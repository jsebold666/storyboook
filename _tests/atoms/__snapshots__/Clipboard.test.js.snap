// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Clipboard> to render 1`] = `
<Clipboard
  onChange={[Function]}
  value="This is text"
>
  <div
    onChange={[Function]}
    onClick={[Function]}
    value="This is text"
  />
</Clipboard>
`;

exports[`Expect <Clipboard> to render properly with empty 1`] = `
<Clipboard
  onChange={[Function]}
  value=""
>
  <div
    onChange={[Function]}
    onClick={[Function]}
    value=""
  />
</Clipboard>
`;
